src/systemctl/systemctl.c

```diff
@@ -975,6 +982,16 @@ static int systemctl_parse_argv(int argc, char *argv[]) {
                         arg_drop_in = optarg;
                         break;
 
+                case ARG_WHEN:
+                        r = parse_timestamp(optarg, &arg_when);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse --when= argument '%s': %m", optarg);
+                        break;
```

Comment:
Please refuse special values: `0` and `USEC_INFINITY`. You can use `timestamp_is_set()`.